internal/providers/amazon/objectstore.go

```diff
@@ -180,19 +184,43 @@ func (s *objectStore) DeleteBucket(bucketName string) error {
 	bucket := &ObjectStoreBucketModel{}
 	searchCriteria := s.searchCriteria(bucketName)
 
-	logger.Info("looking for bucket")
+	logger.Infof("looking up bucket %s", bucketName)
 
 	if err := s.db.Where(searchCriteria).Find(bucket).Error; err != nil {
 		if err == gorm.ErrRecordNotFound {
 			return bucketNotFoundError{}
 		}
 	}
 
-	logger.Info("deleting bucket")
+	if err := s.deleteFromProvider(bucket); err != nil {
+		if !s.force {
+			// if delete is not forced return here
+			return err
+		}
+	}
+
+	db := s.db.Delete(bucket)
+	if db.Error != nil {
+		return s.deleteFailed(bucket, db.Error)
+	}
+
+	return nil
+}
+
+func (s *objectStore) deleteFromProvider(bucket *ObjectStoreBucketModel) error {
+	logger := s.getLogger().WithField("bucket", bucket.Name)
+	logger.Infof("deleting bucket %s on provider", bucket.Name)
+	// todo the assumption here is, that a bucket in 'ERROR_CREATE' doesn't exist on the provider
+	// todo however there might be -presumably rare cases- when a bucket in 'ERROR_DELETE' that has already been deleted on the provider
+	if bucket.Status == providers.BucketCreateError {
+		logger.Debug("bucket doesn't exist on provider")
+		return nil
+	}
+
 	bucket.Status = providers.BucketDeleting
-	err := s.db.Save(bucket).Error
-	if err != nil {
-		return errors.Wrap(err, "could not create AWS object storage client")
+	db := s.db.Save(bucket)
+	if db.Error != nil {
+		return fmt.Errorf("could not update bucket: %s", bucket.Name)
```

Comment:
use `errors.Wrap` should be used here to keep the details of the original error.